### PR TITLE
[4.0] Fix radio btn group

### PR DESCRIPTION
**Pull request by Bakual**
Currently, radios with the class "btn-group" are broken in J4 as you can't change the value at all. The radio is just unresponsive to your clicking.

### Summary of Changes
* Removes the special JS done in the Cassiopeia and Atum templates. Since Bootstrap is loaded anyway and it offers the same functionality, we can use just that one for now (until it gets eventually converted to a CE).
* Adjusts the JLayout for the radio formfield so it works with the Bootstrap code.
Thanks to @C-Lodder for that work!

### Testing Instructions
To test, you can add the following code to the config.xml of any extension. Eg to /administrator/components/com_content/config.xml. Just put it right after the `<config>` tag.
````
	<fieldset
		name="test"
		label="Test"
	>
		<field
			name="plain"
			type="radio"
			class=""
			default="1"
		>
			<option value="1">Foo</option>
			<option value="0">Bar</option>
		</field>

		<field
			name="btn-group"
			type="radio"
			class="btn-group"
			default="1"
		>
			<option value="1">Foo</option>
			<option value="0">Bar</option>
		</field>

		<field
			name="btn-group-yesno"
			type="radio"
			class="btn-group btn-group-yesno"
			default="1"
		>
			<option value="1">JYES</option>
			<option value="0">JNO</option>
		</field>

		<field
			name="btn-group-yesno reordered"
			type="radio"
			class="btn-group btn-group-yesno"
			default="1"
		>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>

		<field
			name="switcher"
			type="radio"
			class="switcher"
			default="1"
		>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>
	</fieldset>
````
Now go to the settings of that component. You will have a new tab with 5 parameters.
Check if the radios work.
Make also sure the plain radios and the switcher work as well.

### Expected result
image
All radios work
We even get some styling as expected, but main point is they work 😄 

### Actual result
image
The marked radios are not working (and styling doesn't exist).

### Documentation Changes Required
None

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@Bakual can you also incorporate the changes in https://github.com/joomla/joomla-cms/pull/19296
So we don’t have a bunch of prs doing a little bit of something over the same files?

**Comment by Bakual** [comment]
@dgt41 Haven't seen that PR before but I like the basic idea to split the layouts. However I need to have a look at it first. Will comment on the other PR with my first thoughts 😄

**Comment by infograf768** [comment]
Except for the density of the Grey colour (too light) when class switcher is used and No is chosen, this works fine here.

**Comment by Bakual** [comment]
I try to not comment on that switcher thing :zipper_mouth_face: 
@infograf768 Can you mark it as successful test then? The switcher colors aren't part of this PR, I just included the switcher so it is verfified it still works.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on e5398e492a426f3f57124113088da5775aa35389<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19326">issues.joomla.org/tracker/joomla-cms/19326</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on e5398e492a426f3f57124113088da5775aa35389<br /><br />Hint: Switch of 

- btn-group
- btn-group-yesno
- btn-group-yesno reordered

works different to Switch "switcher" cause User has to click on "Foo" or "Bar"; at "switcher" User click on green Field to switch from "Yes" to "No" and vice versa.

But i guess this is out of Scope of PR so a successfully test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19326">issues.joomla.org/tracker/joomla-cms/19326</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

File: layouts/joomla/form/field/radio.php [hunk 1]
```diff
@@ -93,36 +92,62 @@
...
 <?php else: ?>
 	<?php // START RADIO TOGGLE ?>
-	<fieldset id="<?php echo $id; ?>" class="<?php echo trim($class . ' radio'); ?>"
-		<?php echo $disabled ? 'disabled' : ''; ?>
-		<?php echo $required ? 'required aria-required="true"' : ''; ?>
-		<?php echo $autofocus ? 'autofocus' : ''; ?>
-		<?php echo $dataToggle; ?>>
+	<?php
+		$isBtnGroup  = strpos(trim($class), 'btn-group') !== false;
+		$isBtnYesNo  = strpos(trim($class), 'btn-group-yesno') !== false;
```
Comment (ReLater) [inline layouts/joomla/form/field/radio.php:97]: I think both trim(...) are not needed here.

Comment (Bakual) [reply layouts/joomla/form/field/radio.php:97]: Since this PR is already RTC, I'd rather leave it there. The trim certainly doesn't hurt.

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
Tests fixed with 550ef7b73c

**Comment by HB12broker** [comment]
Every successful business men and women keeps the growth of their capital through investment, to also be the most riches person in the world investment must be your first priority. Today, you can be an investor with the help of a manager that knows how to control capitals in trading without lost and apply for loan. Trade with Optionfield company today and have your agent to manage your account, for enquiry email on: houngbrowne@gmail.com <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19326">issues.joomla.org/tracker/joomla-cms/19326</a>.</sub>

**Pull request merged**